lang/cs/Org.Apache.REEF.Wake/Time/Time.cs

```diff
@@ -29,6 +29,11 @@ public abstract class Time : IComparable<Time>
     {
         public Time(long timeStamp)
         {
+            if (timeStamp < 0)
+            {
+                throw new ArgumentException("Time must have timeStamp of at least 0.");
```

Comment:
I think we should just stick with `long` right now to avoid breaking the API. If we end up implementing `ITime`, we can then use `ulong`.